src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/JdepsParser.kt

```diff
@@ -57,51 +53,29 @@ internal class JdepsParser private constructor(
     }
   }
 
-  private enum class Mode {
-    COLLECT_DEPS,
-    DETERMINE_JDK,
-    COLLECT_PACKAGES_JDK8,
-    COLLECT_PACKAGES_JDK9
+  private fun addModuleDependency(module: String, jarFile: String) {
+    val entry = moduleDeps.computeIfAbsent(module) {
+      mutableListOf<String>()
+    }
+    entry.add(jarFile)
   }
 
-  // maybe simplify this by tokenizing on whitespace and arrows.
   private fun processLine(line: String) {
-    val trimmedLine = line.trim { it <= ' ' }
-    when (mode) {
-      Mode.COLLECT_DEPS -> if (!line.startsWith(" ")) {
-        val parts = line.split(" -> ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
-        if (parts.size == 2) {
-          if (parts[0] != filename) {
-            throw RuntimeException("should only get dependencies for dep: $filename")
-          }
-          consumeJarLine(parts[1], Deps.Dependency.Kind.EXPLICIT)
-        }
-      } else {
-        mode = Mode.DETERMINE_JDK
-        processLine(line)
-      }
-      Mode.DETERMINE_JDK -> {
-        mode = Mode.COLLECT_PACKAGES_JDK8
-        if (!line.endsWith(packageSuffix)) {
-          mode = Mode.COLLECT_PACKAGES_JDK9
-        }
-        processLine(line)
-      }
-      Mode.COLLECT_PACKAGES_JDK8 -> when {
-        trimmedLine.endsWith(packageSuffix) -> packages.add(
-          trimmedLine.substring(
-            0,
-            trimmedLine.length - packageSuffix.length
-          )
-        )
-        trimmedLine.startsWith("-> ") -> {
-          // ignore package detail lines, in the jdk8 format these start with arrows.
+    val parts = line.split(" -> ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
```

Comment:
Is `" -> ".toRegex()` interned? This feels like it should be constant-ish, especially if it's going to do a Regex compilation on every line.